lib/src/empty_placeholder.dart

```diff
@@ -0,0 +1,255 @@
+import 'package:empty_placeholder_view/src/empty_placeholder_refresh_button.dart';
+import 'package:empty_placeholder_view/src/spin_kit_loader.dart';
+import 'package:empty_placeholder_view/src/values/enums.dart';
+import 'package:empty_placeholder_view/src/values/strings.dart';
+import 'package:empty_placeholder_view/src/vector_asset.dart';
+import 'package:flutter/material.dart';
+
+import 'extensions.dart';
+
+class EmptyListPlaceholder extends StatelessWidget {
+  final ApiState? state;
+  final VoidCallback? onButtonClick;
+  final String? placeholderImage;
+  final String? darkPlaceholderImage;
+  final EdgeInsets? extraPadding;
+  final String? buttonText;
+  final double? buttonWidth;
+  final Color? buttonColor;
+  final Color? buttonTextColor;
+  final bool hideButton;
+  final String? loadingTitle;
+  final TextStyle? titleTextStyle;
+  final String? loadingSubtitle;
+  final TextStyle? subTitleTextStyle;
+  final String? errorTitle;
+  final String? errorSubtitle;
+  final String? emptyTitle;
+  final String? emptySubtitle;
+  final LoaderName? loadingWidgetName;
+  final Widget? loadingWidget;
+  final Color? loadingColor;
+  final double? loadingSize;
+  final String? animationName;
+  final bool isLoadingOnTop;
+
+  const EmptyListPlaceholder({
+    Key? key,
+    @required this.state,
+    this.onButtonClick,
+    this.placeholderImage,
+    this.darkPlaceholderImage,
+    this.extraPadding,
+    this.buttonText,
+    this.buttonWidth,
+    this.buttonColor,
+    this.buttonTextColor,
+    this.hideButton = false,
+    this.loadingTitle,
+    this.titleTextStyle,
+    this.loadingSubtitle,
+    this.subTitleTextStyle,
+    this.errorTitle,
+    this.errorSubtitle,
+    this.emptyTitle,
+    this.emptySubtitle,
+    this.loadingWidgetName,
+    this.loadingWidget,
+    this.loadingColor,
+    this.loadingSize,
+    this.animationName,
+    this.isLoadingOnTop = false,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    String? title;
+    String? subtitle;
+
+    switch (state) {
+      case ApiState.loading:
+        {
+          title = loadingTitle ?? AppStrings.holdOn;
+          subtitle = loadingSubtitle;
+        }
+        break;
+
+      case ApiState.success:
+        {
+          title = emptyTitle ?? AppStrings.unavailable;
+          subtitle = emptySubtitle;
+        }
+        break;
+
+      case ApiState.error:
+        {
+          title = errorTitle ?? AppStrings.oops;
+          subtitle = errorSubtitle ?? AppStrings.somethingWentWrong;
+        }
+        break;
+
+      // ApiState.idle
+      default:
+        {
+          title = '';
+          subtitle = '';
+        }
+        break;
+    }
+    return SizedBox.expand(
+      child: Container(
+        padding: const EdgeInsets.all(40),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
```

Comment:
crossAxisAlignment of the column has default value as center so no requirement to mention it explicitly